javascript/ql/lib/semmle/javascript/frameworks/NoSQL.qll

```diff
@@ -537,14 +542,22 @@ private module Mongoose {
       // NB: the complete information is not easily accessible for deeply chained calls
       f.getQueryArgument().getARhs() = result
     }
+
+    override DataFlow::Node getAResult() {
+      result = this.getCallback(this.getNumArgument() - 1).getParameter(1)
+    }
   }
 
-  class ExplicitQueryEvaluation extends DatabaseAccess {
+class ExplicitQueryEvaluation extends DatabaseAccess {
     ExplicitQueryEvaluation() {
       // explicit execution using a Query method call
       Query::getAMongooseQuery().getMember(["exec", "then", "catch"]).getACall() = this
     }
 
+    override DataFlow::Node getAResult() {
+      result = this.(DataFlow::CallNode).getCallback(_).getParameter(1)
+    }
```

Comment:
2 things:  

1) You can move `DataFlow::CallNode` to the `extends` clause, that way you don't have to do an line cast here. 
2) It looks suspicious to me that you are selecting the second parameter of every callback.  
    You are saying that `foo` in `query.then((foo) => ...)` is **not** a result?

Reply:
1. is locally resolved, but 2. fixed this (good catch!) and added a unit test case for it, too.